thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java

```diff
@@ -249,6 +249,12 @@ public DetectionPipelineResult run() throws Exception {
       checkEarlyStop(totalNestedMetrics, successNestedMetrics, i, lastException);
       MetricEntity metric = nestedMetrics.get(i);
       try {
+        if (Thread.interrupted()) {
```

Comment:
Mind add some comment here specific when will the thread get interrupted? For example, when timing out or shutdown.

Reply:
Just added the comment.

Reply:
Please also add more context like below "Detection with multiple dimensions may take a significantly long time, which would be cancelled from TaskDriver."